### PR TITLE
Copyable state

### PR DESCRIPTION
**Pull request by K20shores**
To fix [255](https://github.com/NCAR/musica/pull/225) in musica, the state needs to be moveable so that store the state properly copies the temporary variables

### CODE REVIEW

**Review by mattldawson — APPROVED** [review]

File: include/micm/solver/state.hpp [hunk 1]
```diff
@@ -59,6 +59,10 @@ namespace micm
     std::size_t number_of_grid_cells_;
     std::unique_ptr<TemporaryVariables> temporary_variables_;
 
+    /// @brief Constructor with parameters
+    /// @param parameters State dimension information
+    State(const StateParameters& parameters);
+
```
Comment (boulderdaze) [inline include/micm/solver/state.hpp:65]: Can't find the definition of this constructor. Is it a placeholder?

File: include/micm/solver/state.hpp [hunk 2]
```diff
@@ -139,6 +182,10 @@ namespace micm
     /// @brief Print state (concentrations) at the given time
     /// @param time solving time
     void PrintState(double time);
+
+    /// @brief Default constructor
+    /// Only defined to be used to create default values in types, but a default constructed state is not useable
+    State();
```
Comment (boulderdaze) [inline include/micm/solver/state.hpp:188]: Would it make more sense to put them in where the `State(...)` constructors are defined?

**Review by boulderdaze — CHANGES_REQUESTED** [review]
Looks good! Just have a couple minor questions

File: include/micm/solver/state.hpp (hunk 2 above)
Comment (K20shores) [reply include/micm/solver/state.hpp:188]: Sure, I can move it back. I had made it private briefly which is why it's here. But that can't work becuase musica does need to make a default value for it to make the MICM object there

File: include/micm/solver/state.hpp (hunk 1 above)
Comment (K20shores) [reply include/micm/solver/state.hpp:65]: it's in the inl file.

Comment (boulderdaze) [reply include/micm/solver/state.hpp:69]: ahh my bad

**Review by boulderdaze — APPROVED** [review]

**Review by sjsprecious — APPROVED** [review]

**Pull request merged**